packages/next/pages/_document.tsx

```diff
@@ -426,11 +445,21 @@ export class NextScript extends Component<OriginProps> {
 
   context!: DocumentComponentContext
 
+  static safariNomoduleFix =
+    '!function(){var e=document,t=e.createElement("script");if(!("noModule"in t)&&"onbeforeload"in t){var n=!1;e.addEventListener("beforeload",function(e){if(e.target===t)n=!0;else if(!e.target.hasAttribute("nomodule")||!n)return;e.preventDefault()},!0),t.type="module",t.src=".",e.head.appendChild(t),t.remove()}}();'
+
```

Comment:
The fix looks pretty cheap so it feels like it could be included by default. Perhaps it could start with a comment like `/* module fix for Safari 10 */` to help developers who explore own sources understand what's going on and easily opt out if they no for sure that their customers won't use this browser.